addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -214,15 +214,19 @@ def _check_unique_vendor_number(self):
         for rec in self.filtered(
                 lambda x: x.name and x.name != '/' and x.is_purchase_document() and x.l10n_latam_use_documents
                             and x.commercial_partner_id):
-            domain = [
-                ('move_type', '=', rec.move_type),
-                # by validating name we validate l10n_latam_document_type_id
-                ('name', '=', rec.name),
-                ('company_id', '=', rec.company_id.id),
-                ('id', '!=', rec.id),
-                ('commercial_partner_id', '=', rec.commercial_partner_id.id),
-                # allow to have to equal if they are cancelled
-                ('state', '!=', 'cancel'),
-            ]
+            domain = rec._get_domain_unique_vendor_number()
```

Comment:
Why do you create this method ?

Reply:
@luistorresm It is probably not necessary to inherit this anymore.  It was taken into account for the _sequence module afaik.